dvc/lock.py

```diff
@@ -114,10 +115,20 @@ def _do_lock(self):
             raise LockError(FAILED_TO_LOCK_MESSAGE)  # noqa: B904
 
     def lock(self):
-        retries = 6
-        delay = DEFAULT_TIMEOUT / retries
-        lock_retry = retry(retries, LockError, timeout=delay)(self._do_lock)
-        lock_retry()
+        """Acquire the lock, either waiting forever, or after default_retries."""
+        default_retries = 6
+        retries = None if self._wait else default_retries
+        delay = DEFAULT_TIMEOUT / default_retries
+        attempts = 0
+
+        while retries is None or attempts < retries:
+            try:
+                self._do_lock()
+                return
+            except LockError:
+                attempts += 1
+                time.sleep(delay)
```

Comment:
Consider re-raising inside `try .. except` instead of raising a new error. We'll lose cause/context in that case.

```suggestion

            try:
                self._do_lock()
                return
            except LockError:
                attempts += 1
                if self._wait or attempts < retries:
                	time.sleep(delay)  
                	continue
                raise  
```